qiskit/quantum_info/states/densitymatrix.py

```diff
@@ -113,12 +113,18 @@ def __eq__(self, other):
             self._data, other._data, rtol=self.rtol, atol=self.atol)
 
     def __repr__(self):
-        text = self.draw('text', prefix="DensityMatrix(")
-        return str(text) + ')'
+        prefix = 'DensityMatrix('
+        pad = len(prefix) * ' '
+        return '{}{},\n{}dims={})'.format(
+            prefix, np.array2string(
+                self._data, separator=', ', prefix=prefix),
+            pad, self._op_shape.dims_l())
 
     def draw(self, output=None, max_size=(16, 16), dims=None, prefix='', **drawer_args):
```

Comment:
Do we need the `dims` and `prefix` kwargs? dims can be gotten from the current object, and I'm not sure prefix is needed any longer if you have output=`repr`.

Reply:
`dims` is a boolean option to control the inclusion of dims or not.
`prefix` allows writing thinks like:
```
Statevector.from_label('0+00').draw('latex', prefix='\\textrm{state}=')
```
<img width="533" alt="Screenshot 2021-02-24 at 10 07 57" src="https://user-images.githubusercontent.com/766693/108976807-30b91d80-7688-11eb-9868-dcfd7322fcb3.png">

If it is not useful, it can be removed.

Reply:
Sounds like they are drawer options of the text output `state_drawer` methods then. They could be added to that function and go under drawer_args.